packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -162,26 +177,21 @@ class _SegmentState<T> extends State<_Segment<T>> with TickerProviderStateMixin<
             child: AnimatedDefaultTextStyle(
               style: DefaultTextStyle.of(context)
                 .style
-                .merge(TextStyle(fontWeight: widget.highlighted ? FontWeight.w500 : FontWeight.normal)),
+                .merge(TextStyle(
+                  fontWeight: widget.highlighted ? FontWeight.w600 : FontWeight.w500,
+                  fontSize: 13,
+                  color: widget.enabled ? null : _kDisabledContentColor,
+                )),
               duration: _kHighlightAnimationDuration,
               curve: Curves.ease,
               child: ScaleTransition(
+                alignment: scaleAlignment,
```

Comment:
Ah so the thumb press animation behaves differently if it's on the leftmost / rightmost segments?

Reply:
Yes!